File: ios/AppInfo.m
//
//  AppInfo.m
//  BgRemover
//
//  AppInfo.swift を React Native に登録するための橋渡し。
//  ReviewManager と同じく RCT_EXTERN_MODULE 方式なので bridging header は不要。
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AppInfo, NSObject)

@end

File: ios/AppInfo.swift
//
//  AppInfo.swift
//  BgRemover
//
//  Info.plist（＝Xcodeの MARKETING_VERSION）からアプリのバージョン文字列を
//  取得して JS へ渡すモジュール。package.json のバージョンとは独立して
//  Xcode 側の値に自動追従させるためのもの。
//

import Foundation
import React

@objc(AppInfo)
class AppInfo: NSObject {

  @objc
  func constantsToExport() -> [String: Any] {
    let version = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String
    return ["version": version ?? ""]
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
